antarest/study/business/model/xpansion_model.py

```diff
@@ -213,3 +220,63 @@ class XpansionSettingsUpdate(XpansionSettings):
         alias="additional-constraints",
         description="Additional constraints file",
     )
+
+
+class XpansionCandidateBase(AntaresBaseModel, populate_by_name=True):
+    name: str
+    link: str
+    annual_cost_per_mw: float = Field(gt=0)
+    unit_size: Optional[float] = Field(default=None, ge=0)
+    max_units: Optional[int] = Field(default=None, ge=0)
+    max_investment: Optional[float] = Field(default=None, ge=0)
+    already_installed_capacity: Optional[int] = Field(default=None, ge=0)
+    # this is obsolete (replaced by direct/indirect)
+    link_profile: Optional[str] = None
+    # this is obsolete (replaced by direct/indirect)
+    already_installed_link_profile: Optional[str] = None
+    direct_link_profile: Optional[str] = None
+    indirect_link_profile: Optional[str] = None
+    already_installed_direct_link_profile: Optional[str] = None
+    already_installed_indirect_link_profile: Optional[str] = None
+
+    @model_validator(mode="after")
+    def validate_model(self) -> "XpansionCandidateBase":
+        possible_format_1 = self.max_investment is None and (self.max_units is not None and self.unit_size is not None)
+        possible_format_2 = self.max_investment is not None and (self.max_units is None and self.unit_size is None)
+
+        if not (possible_format_1 or possible_format_2):
+            raise BadCandidateFormatError(
+                "The candidate is not well formatted."
+                "\nIt should either contain max-investment or (max-units and unit-size)."
+            )
+
+        return self
+
+    @field_validator("name", mode="before")
+    def validate_name(cls, name: str) -> str:
+        # The name is written directly inside the ini file so a specific check is performed here
+        if name.strip() == "":
+            raise CandidateNameIsEmpty()
+
+        illegal_name_characters = [" ", "\n", "\t", "\r", "\f", "\v", "-", "+", "=", ":", "[", "]", "(", ")"]
+        for char in name:
+            if char in illegal_name_characters:
+                raise IllegalCharacterInNameError(f"The character '{char}' is not allowed in the candidate name")
+
+        return name
+
+    @field_validator("link", mode="before")
+    def validate_link(cls, link: str) -> str:
+        if " - " not in link:
+            raise WrongLinkFormatError(f"The link must be in the format 'area1 - area2'. Currently: {link}")
+        return link
+
+
+class XpansionCandidateInternal(XpansionCandidateBase, alias_generator=to_kebab_case):
```

Comment:
we'd better use plain names, not "Internal": `XpansionCandidateUpdate` and `XpansionCandidateCreation` for example.
The 2 differ at least by the fact that name is optional for the update but not the creation.
Also, link and annual_cost_per_mw should be optionals, I guess.

Reply:
As discussed, the current behaviour is to actually REPLACE the whole candidate, because sometimes we want to remove fields. With an update class as usual, this would not be possible.
Hence we decided to renamed the command as `ReplaceXpansionCandidate` because it's different from other update commands.